pkg/metrics/http.go

```diff
@@ -125,3 +129,36 @@ type TargetInfo struct {
 	ScrapeDuration   int64         `json:"scrape_duration_ms"`
 	ScrapeError      string        `json:"scrape_error"`
 }
+
+// PushMetricsHandler provides a way to POST data directly into
+// an instance's WAL.
+func (a *Agent) PushMetricsHandler(w http.ResponseWriter, r *http.Request) {
+	// Get instance name.
+	instanceName, err := getInstanceName(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Get the metrics instance and serve the request.
+	managedInstance, err := a.InstanceManager().GetInstance(instanceName)
+	if err != nil || managedInstance == nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	handler := remote.NewWriteHandler(a.logger, managedInstance)
+	handler.ServeHTTP(w, r)
+}
```

Comment:
IMO this is something that's so obviously correct at a glance that we might not need the test, especially since the test is pretty large. Up to you to keep/remove though :)

Reply:
I didn't know what the team would prefer, but I'd also be inclined to remove it, since it's mostly reiterating the upstream Prometheus test.

I'll remove it and see if anyone else raises a concern.